paparazzi/src/main/java/app/cash/paparazzi/internal/apng/APNGReader.kt

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.paparazzi.internal.apng
+
+import app.cash.paparazzi.internal.apng.PngConsts.Header
+import app.cash.paparazzi.internal.apng.PngConsts.Header.ACTL
+import app.cash.paparazzi.internal.apng.PngConsts.Header.FCTL
+import app.cash.paparazzi.internal.apng.PngConsts.Header.FDAT
+import app.cash.paparazzi.internal.apng.PngConsts.Header.IDAT
+import app.cash.paparazzi.internal.apng.PngConsts.Header.IEND
+import app.cash.paparazzi.internal.apng.PngConsts.Header.IHDR
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_COLOR_TYPE_RGB
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_COLOR_TYPE_RGBA
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_SIG
+import okio.Buffer
+import okio.BufferedSource
+import okio.FileSystem
+import okio.Path.Companion.toPath
+import okio.buffer
+import okio.internal.commonToUtf8String
+import java.awt.image.BufferedImage
+import java.awt.image.ColorModel
+import java.awt.image.DataBufferInt
+import java.awt.image.Raster
+import java.io.Closeable
+import java.io.File
+import java.lang.IllegalStateException
+import java.util.zip.CRC32
+import java.util.zip.Inflater
+
+/**
+ * Provides a simple sub-spec implementation for reading APNG files. Only handles
+ * chunks which are implemented by the writer.
+ */
+internal class APNGReader(file: File) : Closeable {
```

Comment:
nit: i think Java convention is to use `Apng` vs `APNG` as a prefix?

```suggestion
internal class ApngReader(file: File) : Closeable {
```